util/tracing/opt_trace.go

```diff
@@ -93,3 +93,34 @@ type CETraceRecord struct {
 	Expr      string
 	RowCount  uint64
 }
+
+// PhysicalOptimizeTraceInfo indicates for the PhysicalOptimize trace information
+type PhysicalOptimizeTraceInfo struct {
+	ID       int                          `json:"id"`
+	TP       string                       `json:"type"`
+	Children []*PhysicalOptimizeTraceInfo `json:"children"`
+	Property string                       `json:"property"`
+
+	// ExplainInfo should be implemented by each implemented LogicalPlan
+	ExplainInfo string `json:"info"`
+
+	BestTask   *TaskInfo   `json:"best"`
+	Candidates []*TaskInfo `json:"candidates"`
+}
```

Comment:
Maybe we should maintain candidates like `map[string][]*TasnkInfo`, thus we can group by all the candidates by property?